ladybug/_datacollectionbase.py

```diff
@@ -407,6 +414,74 @@ def are_collections_aligned(data_collections, raise_exception=True):
+                more numerical values as input.
+            data_collections: A list with a length equal to the number of arguments
+                for the function. Items of the list can be either Data Collections
+                or individual values to be used at each datetime of other collections.
+            result_type: An instance of a Ladybug data type that describes the resulting
+                numerical values.
+            result_unit: The units of the resulting numerical values.
+
+        Return:
+            A Data Collection with the results function. If all items in this list of
+            data_collections are individual values, only a single value will be returned.
+
+        Usage:
+            from ladybug.datacollection import HourlyContinuousCollection
+            from ladybug.epw import EPW
+            from ladybug.psychrometrics import humid_ratio_from_db_rh
+            from ladybug.datatype.percentage import HumidityRatio
+
+            epw_file_path = './epws/denver.epw'
+            denver_epw = EPW(epw_file_path)
+            pressure_at_denver = 85000
+            hr_inputs = [denver_epw.dry_bulb_temperature,
+                         denver_epw.relative_humidity,
+                         pressure_at_denver]
+            humid_ratio = HourlyContinuousCollection.compute_function_aligned(
+                humid_ratio_from_db_rh, hr_inputs, HumidityRatio(), 'fraction')
+            # humid_ratio will be a Data Colleciton of humidity ratios at Denver
+        """
+        # first check the input collections or individual values
+        data_c = []
+        numerical_vals = []
+        for func_input in data_collections:
+            if isinstance(func_input, BaseCollection):
+                data_c.append(func_input)
+            else:
+                try:
+                    numerical_vals.append(float(func_input))
+                except ValueError:
+                    raise TypeError('Expected a number or a Data Colleciton. '
+                                    'Got {}'.format(type(func_input)))
+
+        # run the function and return the result
+        if len(data_c) == 0:
+            # just inidivdual numerical values
+            return funct(*numerical_vals)
+        else:
+            # data collections
+            aligned_collections = []
+            for func_input in data_collections:
+                if isinstance(func_input, BaseCollection):
+                    aligned_collections.append(func_input)
+                else:
+                    aligned_collections.append(
+                        data_c[0].get_aligned_collection(float(func_input)))
+            BaseCollection.are_collections_aligned(aligned_collections)
+            result_values = []
+            for i in xrange(len(data_c[0].values)):
+                input_list = [coll[i] for coll in aligned_collections]
+                result_values.append(funct(*input_list))
+            return data_c[0].get_aligned_collection(
```

Comment:
I'm not sure why you have to create a list and then use another method to create a datacollection. Can't you just create a datacollection and add the values to it during the loop?

Reply:
I don't have to do it this way and I guess you are right that the other way is more readable. So I have changed it over to be that way in the commit I am about to push.

However, both have similar performance because the current DataCollections don't allow you to append values.  Doing so would mean that you have to include a datetime for this new value or, in the case of a continuous collection, it would break the relationship between the number of values of the collection and the number of values permitted by the header analysis_period.  So, when I first create the collection, it actually comes with placeholder values that either have datetimes or have times that are fully described by the header analysis_period.  So, rather than appending values to the collection, I instead individually set the placeholder values to be the ones that I want.